### PR TITLE
Allow missing given names in AliasManagementService

### PR DESCRIPTION
**Pull request by awead**
If no given name is provided, a new record will be created using only
the surname. Searching for a nil given name returns only records without
that given name present, avoiding the issue of two existing persons with
the same surname, but one with a missing given name.

### CODE REVIEW

File: spec/services/alias_management_service_spec.rb [hunk 1]
```diff
@@ -76,22 +76,35 @@
   end
 
   context 'when the person exists but the alias does not' do
-    before { Alias.destroy_all }
+    let(:new_alias) { Alias.where(display_name: 'Capt. Jack Sparrow').first }
 
-    context 'with all the required parameters' do
+    before do
+      Alias.destroy_all
+      if Person.where(given_name: 'Johnny', sur_name: 'Depp').empty?
```
Comment (carolyncole) [inline spec/services/alias_management_service_spec.rb]: Why is this needed?  Couldn't you just reference you let?  `person`

**Review by carolyncole — CHANGES_REQUESTED** [review]
Other than the strance create in the before it is all looking good.

File: spec/services/alias_management_service_spec.rb (hunk 1 above)
Comment (awead) [reply spec/services/alias_management_service_spec.rb]: @cam156 this is a "find or create" scenario. `let` was creating multiple Johnny Depps. I wanted to ensure there was only one.

Comment (carolyncole) [reply spec/services/alias_management_service_spec.rb]: @awead you could set the id in the let statement

Comment (awead) [reply spec/services/alias_management_service_spec.rb]: @cam156 possibly. "Let" me give that a try...

**Comment by awead** [comment]
@cam156 I pushed up a change. Providing the id in the create method wasn't working, it was just making a new object. I reorganized the create/update process so that there will always be one of each record type.
